uefi/src/boot.rs

```diff
@@ -120,28 +120,43 @@ pub unsafe fn raise_tpl(tpl: Tpl) -> TplGuard {
     }
 }
 
-/// Allocates memory pages from the system.
+/// Allocates a consecutive set of memory pages using the UEFI allocator.
 ///
-/// UEFI OS loaders should allocate memory of the type `LoaderData`.
+/// The caller is responsible to free the memory using [`free_pages`].
+///
+/// # Arguments
+/// - `allocation_type`: The [`AllocateType`] to alter the allocation strategy.
+/// - `memory_type`: The [`MemoryType`] used to persist the allocation in the
+///   UEFI memory map. Typically, UEFI OS loaders should allocate memory of
+///   type [`MemoryType::LOADER_DATA`].
+///- `size`: Amount of bytes to allocate.
+///
+/// # Safety
+/// Using this function is safe but reading on initialized memory is not.
```

Comment:
```suggestion
/// Using this function is safe, but it returns a raw pointer to uninitialized memory. The memory must be initialized before creating a reference to it.
```

(Reference: https://doc.rust-lang.org/nomicon/unchecked-uninit.html "It is illegal to construct a reference to uninitialized data")

Reply:
Yikes, typos/weird wording from my side. 

Thanks